jenkins-plugin/pom.xml

```diff
@@ -187,6 +187,7 @@
       <groupId>io.jenkins.plugins</groupId>
       <artifactId>h2-api</artifactId>
       <version>${jenkins-plugin-h2.version}</version>
+      <optional>true</optional>
```

Comment:
This will still cause the H2 plugin to be installed if you install `pipeline-maven` from the UC. This does not seem right.

I think what we should be doing here is to depend on the `database` plugin. (Hard dep, no need for `optional`, since by itself it does nothing.) At runtime you can check if there is a database registered. If so, use it. If not, use the null DAO and warn appropriately. Anyone who wishes to use H2 can do so by installing `database-h2` and configuring a local H2 DB.

Reply:
Aggreed, but already discussed. Switching to ` database`  plugin is on my TODO list (a JIRA issue request support for SQL Server, I want to solve this issue with this switch) but is quite a huge change. I start to look, it will need some automated migration stuff, ` database`  plugin has its own internal schema, which is not directly the same as the one in our current MySQL/PostgreSQL databases.

So must be a dedicated change, in a second time